### PR TITLE
Finetuning Fixes

### PR DESCRIPTION
**Pull request by Satrat**
* Only compare non-padding tokens in distillation to match with paper
* Remove padding_mask column, using attention_mask instead to simplify things
* Mask out prompt in labels to match llama-recipes implementation, all datasets that use prompts add a "prompt" column
* Use DefaultDataCollator instead of CausalLM one, otherwise HuggingFace overwrites our labels!

### Metrics on Llama7b Distillation
* Winogrande (32): 73.56%
* Arc Challenge (16): 50.09%
* TruthfulQA (32): 43.25% 
* GSM8k (16): 

Full Results: 
SparseML: https://docs.google.com/spreadsheets/d/1-AMyNT8oY1bDmtkRm3llnhNmK6FMIX7mxYlgcWC4O6c/edit#gid=2087112309
Llama Recipes: https://docs.google.com/spreadsheets/d/1qeEi9ICbu9hkwpnFReOh1jbXShj0dOGYbaFb3M2AVi4/edit#gid=0

### CODE REVIEW

File: src/sparseml/transformers/finetune/data/base.py [hunk 1]
```diff
@@ -172,51 +184,22 @@ def label_fn(data):
                 desc="Grouping text",
             )
 
+        if isinstance(dataset, IterableDataset):
+            # so we can get column names from streamed_datasets
+            dataset = dataset._resolve_features()
```
Comment (rahul-tuli) [inline src/sparseml/transformers/finetune/data/base.py:189]: nice!

**Review by rahul-tuli — APPROVED** [review]
Beautiful code!

**Review by rahul-tuli — COMMENTED** [review]
Some tests are failing though :(

**Review by bfineran — APPROVED** [review]
